stem/model.py

```diff
@@ -67,114 +67,101 @@ def __del__(self):
-        # rail_pads_model_part.get_geometry_from_geo_data(self.gmsh_io.geo_data, rail_pads_name)
-        # rail_pads_model_part.material = StructuralMaterial(name=rail_pads_name, material_parameters=rail_pad_parameters)
-        #
-        # # add physical group to gmsh
-        # rail_constraint_name = f"constraint_{rail_name}"
-        # rail_constraint_geometry_ids = self.gmsh_io.geo_data["physical_groups"][rail_name]["geometry_ids"]
-        # self.gmsh_io.add_physical_group(f"constraint_{rail_name}", 1, rail_constraint_geometry_ids)
-        #
-        # # create model part
-        # constraint_model_part = ModelPart(rail_constraint_name)
-        #
-        # # retrieve geometry from gmsh and add to model part
-        # constraint_model_part.get_geometry_from_geo_data(self.gmsh_io.geo_data, rail_constraint_name)
-        #
-        # # add displacement_constraint in x and z direction
-        # constraint_model_part.parameters = DisplacementConstraint(active=[True, True, True],  is_fixed=[True, False, True],
-        #                                                value=[0, 0, 0])
-
-        # _______ WIP______________________________________________________________
+        sleeper_model_part = BodyModelPart(sleeper_name)
+        sleeper_model_part.get_geometry_from_geo_data(self.gmsh_io.geo_data, sleeper_name)
+
+        # create rail pad geometries
+        rail_pad_line_ids_aux = [self.gmsh_io.make_geometry_1d((top_coordinates, bot_coordinates))
+                                 for top_coordinates, bot_coordinates in zip(rail_global_coords, sleeper_global_coords)]
+
+        rail_pad_line_ids = [ids[0] for ids in rail_pad_line_ids_aux]
+
+        self.gmsh_io.add_physical_group(rail_pads_name, 1, rail_pad_line_ids)
+
+        # create rail, sleeper, and rail_pad body model parts
+        sleeper_model_part.material = StructuralMaterial(name=sleeper_name, material_parameters=sleeper_parameters)
+
+        rail_pads_model_part = BodyModelPart(rail_pads_name)
+        rail_pads_model_part.get_geometry_from_geo_data(self.gmsh_io.geo_data, rail_pads_name)
+        rail_pads_model_part.material = StructuralMaterial(name=rail_pads_name, material_parameters=rail_pad_parameters)
+
+        # add physical group to gmsh
+        rail_constraint_name = f"constraint_{rail_name}"
+        rail_constraint_geometry_ids = self.gmsh_io.geo_data["physical_groups"][rail_name]["geometry_ids"]
+        self.gmsh_io.add_physical_group(f"constraint_{rail_name}", 1, rail_constraint_geometry_ids)
+
+        # create model part
+        constraint_model_part = ModelPart(rail_constraint_name)
+
+        # retrieve geometry from gmsh and add to model part
+        constraint_model_part.get_geometry_from_geo_data(self.gmsh_io.geo_data, rail_constraint_name)
+
+        # add displacement_constraint in the non-vertical directions
+        constraint_model_part.parameters = DisplacementConstraint(active=[True, True, True],
```

Comment:
we should use non-reflecting BC 
also the user might want to change BC . this should be an optional parameter on the track definition

Reply:
this BC is for the non vertical direction. Such that the track doesnt fall over (railpads should only move up and down, not sideways). No boundary conditions have been implemented at the end of the track, but users can add a NodalConcentrated element with a certain damping parameter at the end of the track)